### ISSUE TITLE
Results after training

### ISSUE THREAD

**Issue by AMagd** [ge-xing/Diff-UNet#12]
First of all, thanks for publicly sharing your paper and source-code. 

I have used train.py to train a model on BraTS2020 dataset, but the results I got mismatches the results reported in the paper, do you have any idea why is that?

I got the following Dice scores (after training for 300 epochs):
wt is 0.8935, tc is 0.7762, et is 0.7637, mean_dice is 0.8111

**Comment by 920232796** [comment]
Can you show the tensorboard training figures? And what the validation dice scores? The mean validation dice should be about 0.87.

<img width="289" alt="image" src="https://user-images.githubusercontent.com/32668889/235795417-7f82e567-a1e4-4375-8a91-7db7abaf7fa8.png">

**Comment by AMagd** [comment]
Thanks for your reply. 

![image](https://user-images.githubusercontent.com/56494952/235856107-e281d61b-f3c2-4530-a41d-52c877507534.png)

1) Here are the tensorboard plots:
![mean_dice](https://user-images.githubusercontent.com/56494952/235855292-a45e36df-80af-4035-8157-4982a0e03c46.PNG)
![wt](https://user-images.githubusercontent.com/56494952/235855351-ebb9c3b4-b5bf-4079-8bc6-fbcc1e017cb0.PNG)
![et](https://user-images.githubusercontent.com/56494952/235855356-a4fbefd7-b612-4c83-b598-32ab8f27b7d1.PNG)
![loss](https://user-images.githubusercontent.com/56494952/235855359-5204bd78-7274-4603-8f70-80f1c68b0a1e.PNG)
![tc](https://user-images.githubusercontent.com/56494952/235855361-ce769519-1dc4-4529-bcd7-e6d5d7dd25d6.PNG)

2) The average test dice score over the 75 sample (by running test.py) is:
wt = 0.9126328895901329
tc = 0.8598519600752457
et = 0.7791602345663959
mean_dice = 0.8505483614105913

The mean dice for the test dataset (0.85) is much better than the (0.81) I got during training

**Comment by 920232796** [comment]
Your training result seems like a bit strange.

**Comment by 920232796** [comment]
My tensorboard result in training is :

<img width="409" alt="image" src="https://user-images.githubusercontent.com/32668889/235981487-5866a35a-2c0d-4742-ac5e-14c13936fba9.png">

**Comment by AMagd** [comment]
In the image in your first reply, I've noticed that the folder name you have is different than mine "diffusion_seg_all_loss_e300_norm_fix...", so I am assuming that this means that the result you are showing are for a different (maybe a bit enhanced) model, is that correct? or are you getting these results from the exact same code published here?

**Comment by 920232796** [comment]
No, the code is same. You maybe need to check the data? Or you can try to train on the BTCV dataset. I can also provide some information for you.

**Comment by AMagd** [comment]
Hmmm strange, I will double check once again then. Thanks a lot :)
